a_tools.latest_data: new feature match_whole_words

chellings opened the pull request:
By setting the new parameter match_whole_words to False, the old behavior (search also inside words) can be activated.

FYI @stephlazar @antsr

chellings committed:
a_tools.latest_data: match only whole words by default
By setting the new parameter match_whole_words to False,
the old behavior (search also inside words) can be activated.

stephlazar commented:
I was the one who added this functionality but I am not a reviewer :D

stephlazar reviewed (approved):
But I approve :p

chellings commented:
I had invited @nathlacroix to review because I was curious on his opinion on match_whole_word=True by default

nathlacroix reviewed (changes requested):
Indeed I don't think this should be the default behavior. If I search for "cphase", I don't want to bother checking whether there is indeed an underscore before and after. Not really sure about the usecase of this feature (for me you can always adapt the label with/without underscores), but I don't think it hurts, as long as it is still considered as an "advanced" feature used when you know what you're doing and why. 
TLNR; would set match_whole_words=False by default, unless the reason of this change is explained more in details

stephlazar commented:
@nathlacroix  I introduced this feature to deal with the problem 'even though i want label="spec_qb1", it would find "spec_qb11", "spec_qb12", ..., "spec_qb17." ' It was a quick fix because we needed it during measurement so anyone has a better solution please feel free to implement it! I'm sure there's better solutions :P

nathlacroix commented:
ok, I see. Still don't think it is a viable long term solution indeed, because we don't have clear rules as for how to format labels. e.g. for instance when doing ssro, I include the list of qubits, which then has a label "ssro_['qb1', 'qb11']". So the hack in place does not really work in that case, unless we define a clear format for the label. 
So for now I would keep this "feature" as optional and not default, while we don't have a long term solution. Let's see who has time to come back to this at some point :D

stephlazar commented:
It cannot be optional (ie match_whole_words=False) on the XLD setup because then many analyses will fail (will find qb11,...,qb17 instead of qb1). So I think we should keep it to match_whole_words=True until we find a better solution. None of the measurements we have done on the XLD so far with this "feature" have failed.

chellings committed:
a_tools.latest_data: changed default value of match_whole_words to False
It was decided during review that the old behavior be the default and the new feature should be activated only manually by setting match_whole_words=True.

chellings commented:
Proposal: what if I implement it in a way that you can set
```a_tools.latest_data_match_whole_words = True```
in you init script and change the default behavior for you setup?

chellings committed:
a_tools.latest_data: make it possible to specify the default behavior on a setup
by putting a_tools.latest_data_match_whole_words = True into the init script
